### PR TITLE
[ntuple] Remove `Warning: Unused class rule: SG::sgkey_t` in builds with `-Dtesting=ON`

### PR DESCRIPTION
**Pull request by jalopezg-r00t**
In one of the RNTuple tests that exercises user-defined types, `SG::sgkey_t` is a typedef for a fundamental type, which causes rootcling to print a warning:
```
Warning: Unused class rule: SG::sgkey_t
```

This is because `#pragma link C++ typedef foo` is only used by rootcling to generate a dictionary for `foo` if it is a typedef for a class/struct.
For other typedefs, either _(1)_ cxxmodules have to be enabled, or _(2)_ cling has to parse the header.

## Changes or fixes:
- Remove the no-op `#pragma link C++ typedef ...`
- We do not rely on cxxmodules to be enabled for this test; make cling parse `CustomStruct.hxx` instead

## Checklist:
- [X] tested changes locally

### CODE REVIEW

File: tree/ntuple/v7/test/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,10 @@ ROOT_ADD_GTEST(ntuple_serialize ntuple_serialize.cxx LIBRARIES ROOTDataFrame ROO
 ROOT_ADD_GTEST(ntuple_types ntuple_types.cxx LIBRARIES ROOTDataFrame ROOTNTuple MathCore CustomStruct)
 ROOT_ADD_GTEST(ntuple_view ntuple_view.cxx LIBRARIES ROOTDataFrame ROOTNTuple MathCore CustomStruct)
 ROOT_ADD_GTEST(ntuple_zip ntuple_zip.cxx LIBRARIES ROOTDataFrame ROOTNTuple MathCore CustomStruct)
+# Some of these tests require cxxmodules support, e.g. for typedef resolution.
```
Comment (Axel-Naumann) [inline tree/ntuple/v7/test/CMakeLists.txt:39]: What happens in the non-cxxmodules case? I.e. what is the typedef resolution failure you're referring to here?

File: tree/ntuple/v7/test/CustomStructLinkDef.h [hunk 2]
```diff
@@ -11,9 +11,9 @@
 #pragma link C++ class DerivedC+;
 
 #pragma link C++ class IAuxSetOption+;
-#pragma link C++ typedef SG::sgkey_t+;
 #pragma link C++ class PackedParameters+;
-#pragma link C++ class PackedContainer<int>+;
+#pragma link C++ class PackedContainer<int, PackedParameters>+;
+#pragma link C++ class PackedContainer<int, void>+;
```
[the remark this answers is not in the document]
Comment (jalopezg-r00t) [reply tree/ntuple/v7/test/CustomStructLinkDef.h:16]: No, it should not be the case; there's a template specialization for `PackedContainer<T, void>` which does not have the `m_params` member.

File: tree/ntuple/v7/test/CMakeLists.txt (hunk 1 above)
Comment (jalopezg-r00t) [reply tree/ntuple/v7/test/CMakeLists.txt:39]: In `RFieldBase::Create(const std::string &fieldName, const std::string &typeName)`, we normalize the `typeName`, which includes resolving typedefs to one of the other types known to RNTuple.  We do that via `TClassEdit::ResolveTypedef()`.

However, in the non-cxxmodules case, information for the typedef is not available to cling, which causes `TClassEdit::ResolveTypedef()` to return the input untouched.  This in turn, causes RNTuple to throw an exception:
```
Field m_sgkey has unknown type SG::sgkey_t (unchecked RResult access!)
```

Of course we could do `gInterpreter->ProcessLine("#include \"foo.h\"")`, but given that this test is based on a use case  by Marcin Nowak (EP-UAT), I think it should be closer to what a user would do/expect.

Comment (Axel-Naumann) [reply tree/ntuple/v7/test/CMakeLists.txt:39]: > I think it should be closer to what a user would do

That depends. These are unit tests, not integration tests, and as such they should probably have an easy, self-contained setup - in contrast to an experiment's use case. We have tests that assert that the injection of reflection data works, that libraries are loaded if data is needed etc. Here, failure should be mostly limited to functionality in RNTuple, and we should provide a concise test that allow us to reduce the debugging complexity. Makes sense?

File: tree/ntuple/v7/test/CustomStructLinkDef.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (Axel-Naumann) [reply tree/ntuple/v7/test/CustomStructLinkDef.h:16]: OK thanks! Is this testing ROOT's reflection system or RNTuple?

File: tree/ntuple/v7/test/CMakeLists.txt (hunk 1 above)
Comment (jalopezg-r00t) [reply tree/ntuple/v7/test/CMakeLists.txt:39]: Make sense.  Thanks!

**Review by Axel-Naumann — APPROVED** [review]

**Pull request merged**